src/parser/tree.ts

```diff
@@ -97,32 +99,55 @@ export function doTree(
 				i18nFunctions[functionName as keyof typeof i18nFunctions]
 
 			const children = raw.children.slice(1, -1)
-			let translationKeyIndex = 0;
+			let translationKeyIndex = 0
 
 			// Get the translation from the arguments (the quoted strings)
 			for (const child of children) {
-				let node = child;
+				let node = child
+				let nodeValue = node.text
 
 				// unwrap the argument node, which is used in PHP.
-				if (child.type === "argument") {
-					if (child.children.length === 0) continue;
-					node = child.children[0];
+				if (child.type === 'argument') {
+					if (child.children.length === 0) continue
+					node = child.children[0]
 				}
 
-				// skip if it's not a string
-				if (node.type !== "string") continue
+				if (node?.type === ',') {
+					// skip the comma between arguments
+					continue
+				}
+
+				if (stringType.includes(node?.type)) {
+					// unquote the strings
+					nodeValue = nodeValue.slice(1, -1)
+				} else {
+					// unexpected node type
+					console.warn(
+						'Unexpected node type: ' +
+							node?.type + // variable_name
+							' is ' +
+							translationKeys[translationKeyIndex] + // in number
+							' for  ' +
+							nodeValue + // for $number
+							' in ' +
+							filepath // in filename.php
+					)
+					// this string is not translatable and should be skipped
+					continue
```

Comment:
I would think we don't skip the single string but the whole translation with a `return`.

Reply:
no it would only exit from the nested function, feel free to update the code... it's "more" correct as you say!